[inductor] Use empty_strided to create output tensors when testing AOTInductor

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #108419
* __->__ #108364
* #108412

Summary: This will fix 3 fail_accuracy failures in HF.

Test Plan:
```
python benchmarks/dynamo/huggingface.py --bfloat16 --accuracy --inference --device cuda --export-aot-inductor --only  T5Small
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @anijain2305

desertfire committed:
[inductor] Use empty_strided to create output tensors when testing AOTInductor

Summary: This will fix 3 fail_accuracy failures in HF.

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Use empty_strided to create output tensors when testing AOTInductor"


Summary: This will fix 3 fail_accuracy failures in HF.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]

angelayi commented on benchmarks/dynamo/common.py:
```diff
@@ -1115,7 +1115,7 @@ class AOTInductorModelCache:
     @classmethod
     def load(cls, model, example_inputs, eager_forward):
-        key = id(model)
+        key = str(model)
```
is this necessary?

jansel replied:
I think a weakref would be better here.

jansel replied:
Actually I am not convinced `str` is correct since doesn't the str not include everything about the model?


Let's change this to a weakref.

desertfire committed:
Update on "[inductor] Use empty_strided to create output tensors when testing AOTInductor"


Summary: This will fix 3 fail_accuracy failures in HF.

Test Plan:
```
python benchmarks/dynamo/huggingface.py --bfloat16 --accuracy --inference --device cuda --export-aot-inductor --only  T5Small
```

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx chenyang78 aakhundov

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge
